src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -46,5 +46,20 @@ class GitVersionPlugin implements Plugin<Project> {
             println project.version
         }
     }
-}
 
+    private File findRootGitDir(File currentRoot) {
+        File gitDir = new File(currentRoot, '.git')
+
+        if (gitDir.exists()) {
+            return gitDir
+        }
+
+        // do not follow to the root directories, return non-existing File object
+        if (currentRoot.parent == '/' || currentRoot.parent == '~') {
```

Comment:
File::getParentFile will return null if there is no parent. That might be a more robust check.